lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -810,15 +839,25 @@ def __init__(
         on_relation = self._charm.on[self._relation_name]
         self.framework.observe(
             on_relation.relation_created,
-            self._on_relation_change,
+            self._on_consumers_changed,
         )
         self.framework.observe(
             on_relation.relation_joined,
-            self._on_relation_change,
+            self._on_consumers_changed,
         )
+        self.framework.observe(
+            on_relation.relation_changed,
+            self._on_relation_changed,
+        )
+
+    def _on_consumers_changed(self, event: RelationEvent) -> None:
+        if not isinstance(event, RelationBrokenEvent):
+            self.update_endpoint(event.relation)
+        self.on.consumers_changed.emit()
```

Comment:
Why no relation id here?

Reply:
The relation id honestly isn't used anywhere else either. In similar API changes, it probably could/should be, so Prometheus/Mimir can be more granular about what alerts they keep/delete, but first we'd need either of them to actually start tracking which alerts correlate to which relations, and have some `alerts(self, relation_id: Optional[int] = None)` call to suss it out.

For just the binary "should remote write be enabled in prom? y/n?" the relation id would be meaningless anyway.